File: Command/PHP/Rotate.php
<?php
require_once('ICommand.php');
require_once('Image.php');

class Rotate implements ICommand {
    private Image $target;
    private int $degree;

    public function __construct(Image $target, int $degree)
    {
        $this->target = $target;
        $this->degree = $degree;
    }

    public function execute() 
    {
        $this->target->rotate($this->degree);
    }
}

File: Command/Dart/Rotate.dart
import 'ICommand.dart';
import 'Image.dart';

class Rotate implements ICommand {
  Image? _target;
  int? _degree;

  Rotate(this._target, this._degree);

  @override
  void execute() {
    var degree = _degree!;

    if (degree != null) {
      _target?.rotate(degree);
    }    
  }
}
